nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/deck_guns.dm

```diff
@@ -487,6 +539,22 @@
 	if(id)
 		addtimer(CALLBACK(src, .proc/link_via_id), 10 SECONDS)
 
+	component_parts = list()
+	component_parts += new/obj/item/ship_weapon/parts/firing_electronics
+	component_parts += new/obj/item/ship_weapon/parts/loading_tray
+	if (max_ammo > 1)
+		if(ispath(text2path("/obj/item/circuitboard/multibarrel_upgrade/_[max_ammo]")))
+			component_parts += text2path("new/obj/item/circuitboard/multibarrel_upgrade/_[max_ammo]")
+		else//this should really never happen unless some major tomfoolery goes on
+			var/obj/item/circuitboard/multibarrel_upgrade/M = new/obj/item/circuitboard/multibarrel_upgrade
```

Comment:
```suggestion
			var/obj/item/circuitboard/multibarrel_upgrade/M = new()
```

Reply:
the created fallback upgrade needs to be inside the turret (or in nullspace), not on the floor